test-network-function/networking/suite.go

```diff
@@ -323,3 +335,99 @@ func testNodePort(env *config.TestEnvironment) {
 		}
 	})
 }
+
+func getDeclaredPortList(container int, podName, podNamespace string, declaredPort map[int]string) map[int]string {
+	protocolName, port, name := "", "", ""
+	ocCommandToExecute := fmt.Sprintf(commandportdeclared, podName, podNamespace, container)
+	res, _ := utils.ExecuteCommand(ocCommandToExecute, ocCommandTimeOut, interactive.GetContext(false))
+	x := strings.Split(res, "\n")
+	for _, i := range x {
+		if strings.Contains(i, "containerPort") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			port = s[0]
+		}
+		if strings.Contains(i, "name") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			name = s[0]
+		}
+		if strings.Contains(i, "protocol") {
+			s := strings.Split(i, ":")
+			protocolName = s[1]
+		}
+		if port == "" || name == "" || protocolName == "" {
+			continue
+		}
+		p, _ := strconv.Atoi(strings.TrimSpace(port))
+		name = strings.TrimSpace(strings.ReplaceAll(name, "\"", ""))
+		protocolName = strings.TrimSpace(strings.ReplaceAll(protocolName, "\"", ""))
+		declaredPort[p] = name + " " + protocolName
+		port, name, protocolName = "", "", ""
+	}
+	return declaredPort
```

Comment:
So `declaredPort` and the below `listeningPort` are maps which are pass-by-reference.  Something bothers me about returning a map that is already declared outside the function.  Could you declare a local-to-the-function variable to return instead of the maps brought in by parameter?

Reply:
I agree that it's better to use a local variable with a different name but - if I do, then I need to merge the 2 maps and that's another code. It's intentionally the thing I send the "declaredPorts" and constantly updating the map.(Same for "listeningPorts")

Reply:
I fixed it.